opentracing-spring-cloud-starter-zipkin/src/main/java/io/opentracing/contrib/spring/cloud/starter/zipkin/ZipkinAutoConfiguration.java

```diff
@@ -68,15 +71,19 @@
 
   @Bean
   @ConditionalOnMissingBean
-  public Sender sender(ZipkinConfigurationProperties properties) {
-    return OkHttpSender.create(properties.getHttpSender().getUrl());
-  }
-
+  public Reporter<Span> reporter(ZipkinConfigurationProperties properties) {
+    String url = properties.getHttpSender().getBaseUrl();
+    if (properties.getHttpSender().getEncoder().equals(JSON_V2) || properties.getHttpSender().getEncoder().equals(PROTO3)) {
+      url += (url.endsWith("/") ? "" : "/") + "api/v2/spans";
```

Comment:
I totally understand where this is coming from (since it's Java and there is unfortunately no easy/cool was to build strings), but IMHO, it would be cleaner with `String.format`

Reply:
This `url = String.format("%s%s", url, (url.endsWith("/") ? "" : "/") + "api/v2/spans");` doesn't seem cleaner to me. I would rather keep the original one

Reply:
It's obviously a matter of preference :)